Upgrade mongo

djsd123 opened the pull request:
We currently have a custom fork of the MongoDB module that we no longer want
to maintain. Moving back to upstream puts us back on the support path
and provides support for the MongoDB 3.

Gated package repository for MongoDB 3.

We have to define package repositories manually due to a bug
in the upstream puppetlabs/mongodb module. https://tickets.puppetlabs.com/browse/MODULES-2532

alexmuller commented:
Broadly this looks good to me.

There are 3 instances of `mongodb::replSet::sets` in hieradata which need to have a lowercase `replset` to match the `mongodb::replset` class in the vendored module, and I'm not sure it's worth changing the replset name to reflect the MongoDB version. Is there a reason to do that?

If I was nitpicking the changes to `hieradata/role.ci-slave.yaml` which support the new Puppet module belong in the first commit rather than the second :)

alexmuller commented on modules/gds_mongodb/manifests/init.pp:
```diff
@@ -1,12 +1,20 @@
 
+  if $enable_mongo3_repo {
+    include ::gds_mongodb::repo::mongodb3
+    include ::mongodb::client
```
I can't figure out why we need to include this new class now, but only for Mongo 3.

djsd123 commented:
Tasks complete:

- lowercased replica set references
- added comment explaining why we include mongo client for version 3
- rebased branch to apply `hieradata/role.ci-slave.yaml` to relevant commit

alexmuller commented:
Thanks!
